templates/terraform/operation.erb

```diff
@@ -0,0 +1,35 @@
+<%
+  product_name = object.__product.name
+-%>
+<%= lines(autogen_notice :go) -%>
+package google
+
+import (
+  "fmt"
+)
+
+type <%= product_name -%>OperationWaiter struct {
+  Config *Config
+  TimeoutMinutes int
+  CommonOperationWaiter
+}
+
+func (w *<%= product_name -%>OperationWaiter) QueryOp() (interface{}, error) {
+  if w == nil {
+    return nil, fmt.Errorf("Cannot query operation, it's unset or nil.")
+  }
+  // Returns the proper get.
+  url := fmt.Sprintf("<%= build_url([object.__product.base_url, async.operation.base_url]).gsub('{{op_id}}', '%s') -%>", w.CommonOperationWaiter.Op.Name)
+  return sendRequestWithTimeout(w.Config, "GET", url, nil, time.Duration(w.TimeoutMinutes) * time.Minute)
```

Comment:
I think that this means the _request_ has a timeout of `TimeoutMinutes`; I'd imagine that we can change this back to `sendRequest` as well as remove the `TimeoutMinutes` value from {{product}}OperationWaiter. `OperationWait` is the function responsible for retrying the operation for however many minutes.

Reply:
I don't entirely follow this.

Reply:
We chatted in person and I understand now!